### PR TITLE
Added Queues and Stacks concept in Rust

### LINKED ISSUE #534: [Rust] Add queues and stacks
# Description

Queues and stacks for Rust aren't presently on Code Thesaurus. This issue is to add that concept to the system.

# Directions

* On the home page, on the right side (the reference sheet side), choose the language and version, and then the concept.
* The page says the data doesn't exist, but clicking the arrow for the template will provide the template and file for you to use.
* Save that file in `/web/thesauruses/[language]/[version]/queues_stacks.json`.
* Fill out all of the `"code"` sections.
* Add in `"not-implemented": "true"` if the concept doesn't exist in the language. You can then delete the `"code"` line.
* Add in `"comment":` if you wish to explain something in the code
* Delete the `"name":` fields before your PR
* Save, commit your code, push to your fork, then make a PR.
* Watch the checks to make sure they all pass. If they don't, click on "Details" and it should tell you what to modify.

# Claiming This Issue

Please leave a comment below so I can assign you to it. This is to prevent multiple people from working on the same issue. 

Check out [CONTRIBUTING.md](https://github.com/codethesaurus/codethesaur.us/blob/main/CONTRIBUTING.md) for more information on how to contribute. 

Check out [https://docs.codethesaur.us/thesaurus/](https://docs.codethesaur.us/thesaurus/) for information on how Code Thesaurus works and how to add to the thesauruses.

# Need Help?

Leave a comment on here with your question, or feel free to ask [@geekygirlsarah](https://twitter.com/geekygirlsarah) on Twitter. You can also email coreteam@codethesaur.us too.

# Thanks!

We really value and appreciate the additions people make to Code Thesaurus, no matter how large or small. So thank you for taking the time to work on this!

### PR DESCRIPTION
**Pull request by chetan-satpute**
## What GitHub issue does this PR apply to?

 

Resolves #534 

## What changed and why?

   
Added Queues and Stacks concept in Rust.

## Checklist

   

- [x] I claimed any associated issue(s) and they are not someone else's
- [x] I have looked at documentation to ensure I made any revisions correctly
- [x] I tested my changes locally to ensure they work
- [ ] (If editing Django) I have added or edited any appropriate unit tests for my changes

## Any additional comments or things to be aware of while reviewing?

   
References used to complete the task: 
[Rust STL collections](https://doc.rust-lang.org/std/collections/index.html)
[Binary Heap](https://doc.rust-lang.org/std/collections/struct.BinaryHeap.html)
[Vec Deque](https://doc.rust-lang.org/std/collections/struct.VecDeque.html)

### CODE REVIEW

File: web/thesauruses/rust/1/queues_stacks.json [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+      "name": "Export a list of all stack items",
+      "not-implemented": true
+    },
+    "stack_clear_all": {
+      "name": "Clear all queue items",
+      "not-implemented": true
+    },
+    "iterate_pointer_data_type": {
+      "name": "Data type of a iterator pointer",
+      "code": ["Iter<'_, T>", "IterMut<'_, T>     // for mutable iterator"]
```
Comment (geekygirlsarah) [inline web/thesauruses/rust/1/queues_stacks.json:280]: Is there an immutable iterator?

**Review by geekygirlsarah — APPROVED** [review]
This looks good! I'm curious about the iterator (it kind of implies there might be an immutable one). I can merge it but will wait to hear back on that.

File: web/thesauruses/rust/1/queues_stacks.json (hunk 1 above)
Comment (chetan-satpute) [reply web/thesauruses/rust/1/queues_stacks.json:280]: - `Iter<'_, T>` is an iterator type that returns immutable references.
- `IterMut<'_, T>` is an iterator type that returns mutable references.

Comment (geekygirlsarah) [reply web/thesauruses/rust/1/queues_stacks.json:280]: Oh, I misread that. That makes sense. 

I'll consider this resolved.

**Review by geekygirlsarah — APPROVED** [review]
I think this is good. I can merge it in. 

Congrats on your first Code Thesaurus contribution! 🎉 

This counts towards Hacktoberfest. Don't forget to register if you're planning to participate.

**Pull request merged**